include/odp/api/spec/schedule.h

```diff
@@ -371,6 +371,31 @@ void odp_schedule_order_lock(uint32_t lock_index);
 void odp_schedule_order_unlock(uint32_t lock_index);
 
 /**
+ * Release existing ordered context lock and acquire a new lock
+ *
+ * This call is valid only when holding an ordered synchronization context.
+ * Release a previously locked ordered context lock and acquire a new ordered
+ * context lock. This call is valid only when there is a single ordered context
+ * lock active. Results are undefined if this call is made when multiple ordered
+ * context locks are acquired in nested fashion.
+ * This call does not release the existing ordered context and only switches
+ * from one ordered lock to another ordered lock within the same ordered
+ * context.
+ *
+ * @param unlock_index	Index of the acquired ordered lock in the current
+ *			context to be released. Results are undefined if the
+ *			ordered lock with the given unlock_index is not held by
+ *			this ordered context.
```

Comment:
"Results are undefined if the
ordered lock with the given unlock_index is not held by
this ordered context."

Better to move this to documentation of unlock() function (not the param, but function text).